sdk/servicebus/azure-servicebus/azure/servicebus/exceptions.py

```diff
@@ -74,145 +68,116 @@ def _error_handler(error):
-        else:
-            error = MessageSendFailed(error=exception)
-        error_need_raise = False
-
-    return error, error_need_close_handler, error_need_raise
-
-
-def _create_servicebus_exception(logger, exception, handler, **kwargs):  # pylint: disable=too-many-statements
-    # transform amqp exceptions into servicebus exceptions
-    error_need_close_handler = True
-    error_need_raise = False
-    if isinstance(exception, _AMQP_CONNECTION_ERRORS):
-        error, error_need_close_handler, error_need_raise = \
-            _handle_amqp_connection_error(logger, exception, handler)
-    elif isinstance(exception, _AMQP_MESSAGE_ERRORS):
-        error, error_need_close_handler, error_need_raise = \
-            _handle_amqp_message_error(logger, exception, **kwargs)
-    elif isinstance(exception, errors.AuthenticationException):
-        logger.info("Authentication failed due to exception: (%r).", exception)
-        error = ServiceBusAuthenticationError(str(exception), error=exception)
-    else:
-        logger.info("Unexpected error occurred (%r). Shutting down.", exception)
-        if kwargs.get("settle_operation"):
-            error = MessageSettleFailed(kwargs.get("settle_operation"), error=exception)
-        elif not isinstance(exception, ServiceBusError):
-            error = ServiceBusError("Handler failed: {}.".format(exception), error=exception)
-        else:
-            error = exception
+        logger.info("Unexpected AMQP error occurred (%r). Handler shutting down.", exception)
+
+    error = error_cls(message=str(exception), error=exception)
+    return error
+
+
+def _handle_amqp_mgmt_error(logger, error_description, condition=None, description=None, status_code=None):
+    if description:
+        error_description += " {}.".format(description)
+
+    raise _handle_amqp_exception_with_condition(
+        logger,
+        condition,
+        description=error_description,
+        exception=None,
+        status_code=status_code
+    )
+
 
-    try:
-        err_condition = exception.condition
-        if err_condition in _NO_RETRY_ERRORS:
-            error_need_raise = True
-    except AttributeError:
-        pass
+def _create_servicebus_exception(logger, exception):
+    if isinstance(exception, AMQPErrors.AMQPError):
+        try:
+            # handling AMQP Errors that have the condition field
+            condition = exception.condition
+            description = exception.description
+            exception = _handle_amqp_exception_with_condition(logger, condition, description, exception=exception)
+        except AttributeError:
+            # handling AMQP Errors that don't have the condition field
+            exception = _handle_amqp_exception_without_condition(logger, exception)
+    elif not isinstance(exception, ServiceBusError):
+        logger.info("Unexpected error occurred (%r). Handler shutting down.", exception)
```

Comment:
```suggestion
        logger.exception("Unexpected error occurred (%r). Handler shutting down.", exception)
```
Assuming this is as exceptional as this seems, may be worth using the above (or error, take your pick.  "exception" is supposed to only be used in an exception handler, which this technically is, if deeply embedded, so should be fine.)